lib/mongo/cluster.rb

```diff
@@ -743,24 +744,32 @@ def add(host, add_options=nil)
     #   server.remove('127.0.0.1:27017')
     #
     # @param [ String ] host The host/port or socket address.
+    # @param [ true | false ] disconnect Whether to disconnect the servers
+    #   being removed. For internal driver use only.
     #
-    # @return [ true|false ] Whether any servers were removed.
+    # @return [ Array<Server> | true | false ] If disconnect is true,
+    #   returns whether any servers were removed. If disconnect is false,
+    #   returns an array of servers that were removed (and should be
+    #   disconnected by the caller).
     #
-    # @since 2.0.0, return value added in 2.7.0
-    def remove(host)
+    # @note The return value of this method is not part of the driver's
+    #   public API.
+    #
+    # @since 2.0.0
+    def remove(host, disconnect: true)
       address = Address.new(host)
       removed_servers = @servers.select { |s| s.address == address }
       @update_lock.synchronize { @servers = @servers - removed_servers }
-      removed_servers.each do |server|
-        if server.connected?
-          server.disconnect!
-          publish_sdam_event(
-            Monitoring::SERVER_CLOSED,
-            Monitoring::Event::ServerClosed.new(address, topology)
-          )
+      if disconnect != false
```

Comment:
Is there a reason to write `if disconnect != false` rather than `if disconnect == true` or just `if disconnect` here?

Reply:
This evaluates to `true` when disconnect is `nil`. `if disconnect == true` evaluates to false for truthy values of disconnect like `0`. What this statement intends to convey is the default is false, and anything other than false is true.

Reply:
I changed the docstring to clarify this.